clang/runtime/dpct-rt/include/dpl_extras/algorithm.h.inc

```diff
@@ -2066,6 +2066,80 @@ inline ::std::pair<Iter1, Iter1> equal_range(_ExecutionPolicy &&policy,
 }
 // DPCT_LABEL_END
 
+// DPCT_LABEL_BEGIN|segmented_reduce_argmin|dpct
+// DPCT_DEPENDENCY_BEGIN
+// DplExtrasIterators|arg_index_input_iterator
+// DplExtrasIterators|key_value_pair
+// DplExtrasVector|is_iterator
+// DPCT_DEPENDENCY_END
+// DPCT_CODE
+template <typename _ExecutionPolicy, typename key_t, typename key_out_t, 
+          typename OffsetIteratorT>
+inline ::std::enable_if_t<dpct::internal::is_iterator<key_t>::value &&
+                          dpct::internal::is_iterator<key_out_t>::value>
+segmented_reduce_argmin(_ExecutionPolicy &&policy, key_t keys_in,
+                        key_out_t keys_out, ::std::int64_t n,
+                        ::std::int64_t nsegments, OffsetIteratorT begin_offsets,
+                        OffsetIteratorT end_offsets) {
+  policy.queue().submit([&](sycl::handler &cgh) {
+    cgh.parallel_for(nsegments, [=](sycl::id<1> i) {
+      ::std::int64_t segment_begin = begin_offsets[i];
+      ::std::int64_t segment_length =
+          ::std::min(n, (::std::int64_t)end_offsets[i]) - segment_begin;
+      if (segment_length <= 0) {
+        *(keys_out + i) = dpct::key_value_pair(
+            ptrdiff_t(1),
+            ::std::numeric_limits<
+                typename ::std::iterator_traits<key_t>::value_type>::max());
+      } else {
+        dpct::arg_index_input_iterator arg_index(keys_in + segment_begin);
+        *(keys_out + i) = *::std::min_element(
+            arg_index, arg_index + segment_length,
+            [](const auto &a, const auto &b) { return a.value < b.value; });
```

Comment:
Can we use std::less, what do you think?

Reply:
We can't just use `::std::less` or omit a comparator for the default because we are comparing dereferenced `dpct::arg_index_input_iterator`  (`dpct::key_value_pair`) by their `value` member variable. 

I suppose we could instead do a `::std::min_element` on `keys_in` using the defaulted `std::less`, and then construct a key value pair using `std::distance`.  I don't have a strong preference either way, but the current implementation seems simpler to me.